### PR TITLE
Add rds_cluster_tf module

### PR DESCRIPTION
**Pull request by Jkovarik**

### CODE REVIEW

File: rds-cluster-tf/terraform.tf.example [hunk 1]
```diff
@@ -0,0 +1,8 @@
+terraform {
+  backend "s3" {
+    region         = "<region>"
+    bucket         = "<state-bucket>>"
```
Comment (botanical) [inline rds-cluster-tf/terraform.tf.example:4]:
```suggestion
    bucket         = "<state-bucket>"
```

File: rds-cluster-tf/main.tf [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+module "rds_cluster" {
+  source = "https://github.com/nasa/cumulus/releases/download/v9.0.0/terraform-aws-cumulus-rds.zip"
+  db_admin_username        = var.db_admin_username
+  db_admin_password        = var.db_admin_password
+  region                   = var.region
+  vpc_id                   = var.vpc_id
+  subnets                  = var.subnets
+  engine_version           = var.engine_version
+  deletion_protection      = true
```
Comment (botanical) [inline rds-cluster-tf/main.tf:23]: It's always set to true here but it's false in the variables? 🤔 Though, this `true` is consistent with what we have on Core

Comment (botanical) [inline rds-cluster-tf/main.tf:23]:
```suggestion
  deletion_protection      = var.deletion_protection
```

File: rds-cluster-tf/variables.tf [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+variable "permissions_boundary_arn" {
+  type    = string
+  default = ""
+}
+
+variable "rds_user_password" {
+  type    = string
+  default = "changeme"
+}
```
Comment (botanical) [inline rds-cluster-tf/variables.tf:77]: newline?

File: rds-cluster-tf/main.tf (hunk 2 above)
Comment (Jkovarik) [reply rds-cluster-tf/main.tf:23]: Good catch!

File: rds-cluster-tf/variables.tf (hunk 3 above)
Comment (Jkovarik) [reply rds-cluster-tf/variables.tf]: 👍🏻 e8ceb2c

**Review by botanical — COMMENTED** [review]
Sorry I didn't catch this earlier but I think we need to add a `vpc_id` variable to `data-persistence-tf`

```
variable "vpc_id" {
  type = string
}
```

**Review by botanical — APPROVED** [review]
Had one comment about adding the `vpc_id` variable but, other than that, looks good ✅

**Comment by Jkovarik** [comment]
Hmm, that's a really good point.  Actually there's a bunch of variables we should add.  This PR was supposed to add the cumulus-rds-tf module, *but* the template deploy repo really needs to be updated with the other changes from this ticket.     Thanks for catching that!

**Comment by Jkovarik** [comment]
Thank you for the review -- this should sort the issues: 8b06a372f3d9816d9776c8df39ae83da17dc8eef

**Review by botanical — APPROVED** [review]
pulled and tested locally and all the changes seem to add up - thanks for fixing this!

**Review by markdboyd — APPROVED** [review]

**Pull request merged**